src/puppetlabs/puppetdb/http/catalogs.clj

```diff
@@ -18,10 +17,16 @@
 (defn catalog-status
   "Produce a response body for a request to retrieve the catalog for `node`."
   [api-version node db url-prefix]
-  (if-let [catalog (with-transacted-connection db
-                     (c/status api-version node url-prefix))]
-    (http/json-response (s/validate catalogs/catalog-query-schema catalog))
-    (http/json-response {:error (str "Could not find catalog for " node)} http/status-not-found)))
+  (let [catalog (first
+                 (eng/stream-query-result :catalogs
+                                          api-version
+                                          ["=" "certname" node]
+                                          {}
+                                          db
+                                          url-prefix))]
+    (if catalog
+      (http/json-response (s/validate catalogs/catalog-query-schema catalog))
+      (http/json-response {:error (str "Could not find catalog for " node)} http/status-not-found))))
```

Comment:
While you're digging in this code, @wkalt raised a point about making these messages consistent in https://github.com/puppetlabs/puppetdb/pull/1419. So "No information is known about [type] [id]". For example:

```
# curl 'http://localhost:8080/pdb/query/v4/factsets/node-1/facts'
{
  "error" : "No information is known about factset node-1"
}
```

Its going to be far easier on the conflicts if this change is made here in your patch.